operator/control-plane/src/market.rs

```diff
@@ -1400,1166 +1433,882 @@ mod tests {
+                    bandwidth: 76,
+                    instance_id: compute_instance_id(0),
+                    eif_url: "https://example.com/enclave.eif".into(),
+                    debug: false,
+                }),
+                TestAwsOutcome::SpinDown(test::SpinDownOutcome {
+                    time: start_time + Duration::from_secs(350),
+                    job: job_id,
+                    instance_id: compute_instance_id(0),
+                    region: "ap-south-1".into(),
+                }),
+            ],
         };
 
-        assert!(!aws.instances.contains_key(&job_num.to_string()))
+        run_test(start_time, logs, job_manager_params, test_results).await;
     }
 
     #[tokio::test(start_paused = true)]
     async fn test_address_whitelisted() {
-        let _ = market::START.set(Instant::now());
+        let start_time = Instant::now();
+        let job_id = format!("{:064x}", 1);
 
-        let job_num = U256::from(1).into();
-        let job_logs: Vec<(u64, Log)> = vec![
-            (0, Action::Open, ("{\"region\":\"ap-south-1\",\"url\":\"https://example.com/enclave.eif\",\"instance\":\"c6a.xlarge\",\"memory\":4096,\"vcpu\":2}".to_string(),31000000000000u64,31000u64,market::now_timestamp().as_secs()).abi_encode_sequence()),
+        let logs = vec![
+            (0, Action::Open, ("{\"region\":\"ap-south-1\",\"url\":\"https://example.com/enclave.eif\",\"instance\":\"c6a.xlarge\",\"memory\":4096,\"vcpu\":2}".to_string(),31000000000000u64,31000u64,0).abi_encode_sequence()),
             (500, Action::Close, [].into()),
-        ].into_iter().map(|x| (x.0, test::get_log(x.1, Bytes::from(x.2), job_num))).collect();
+        ];
 
-        let start_time = Instant::now();
-        // pending stream appended so job stream never ends
-        let job_stream = std::pin::pin!(tokio_stream::iter(job_logs.into_iter())
-            .then(|(moment, log)| async move {
-                let delay = start_time + Duration::from_secs(moment) - Instant::now();
-                sleep(delay).await;
-                log
-            })
-            .chain(tokio_stream::pending()));
-        let mut aws: TestAws = Default::default();
-        let res = market::job_manager_once(
-            job_stream,
-            &mut aws,
-            market::JobId {
-                id: job_num.encode_hex_with_prefix(),
+        let job_manager_params = JobManagerParams {
+            job_id: market::JobId {
+                id: job_id.clone(),
                 operator: "abc".into(),
                 contract: "xyz".into(),
                 chain: "123".into(),
             },
-            &["ap-south-1".into()],
-            300,
-            &test::get_rates(),
-            &test::get_gb_rates(),
-            &Vec::from([
+            allowed_regions: vec!["ap-south-1".to_owned()],
+            address_whitelist: vec![
                 "0x0000000000000000000000000f5f91ba30a00bd43bd19466f020b3e5fc7a49ec".to_string(),
```

Comment:
we can parameterize this address during test setup. And provide the same to test::getLogs(). It was't clear what value should this address match to, since getLogs uses some default address.

Reply:
get_log is actually quite messy, it uses the hardcoded address for everything from emitting contract to sender to depositors, should probably rework that